utils/types/servermode/servermode.go

```diff
@@ -20,9 +21,14 @@ func (m Ack) Mode() Mode {
 	return m.mode
 }
 
-func WithACK(mode Mode, ack func()) Ack {
+func (m Ack) Workspaces() string {
+	return m.workspaces
+}
```

Comment:
I would create a separate struct for Workspaces configs, also I think as soon as we get it from etcd we need to split them into an array.